client/layout/index.jsx

```diff
@@ -332,82 +321,86 @@ class Layout extends Component {
-		const oauth2Client = getCurrentOAuth2Client( state );
-		const wccomFrom = currentQuery?.[ 'wccom-from' ];
-		const isEligibleForJITM = [
-			'home',
-			'stats',
-			'plans',
-			'themes',
-			'plugins',
-			'comments',
-		].includes( sectionName );
-		const sidebarIsHidden = ! secondary || isWcMobileApp();
-		const chatIsDocked = ! [ 'reader', 'theme' ].includes( sectionName ) && ! sidebarIsHidden;
-
-		const isEditor = getSectionName( state ) === 'gutenberg-editor';
-		const isCheckout = getSectionName( state ) === 'checkout';
-		const userAllowedToHelpCenter = shouldShowHelpCenterToUser( getCurrentUserId( state ) );
-
-		const disableFAB =
-			( ( isEditor && config.isEnabled( 'editor/help-center' ) ) ||
-				( isCheckout && config.isEnabled( 'checkout/help-center' ) ) ) &&
-			userAllowedToHelpCenter;
+	connect(
+		( state, { currentSection, currentRoute, currentQuery, secondary } ) => {
+			const sectionGroup = currentSection?.group ?? null;
+			const sectionName = currentSection?.name ?? null;
+			const siteId = getSelectedSiteId( state );
+			const sectionJitmPath = getMessagePathForJITM( currentRoute );
+			const isJetpackLogin = currentRoute.startsWith( '/log-in/jetpack' );
+			const isJetpack =
+				( isJetpackSite( state, siteId ) && ! isAtomicSite( state, siteId ) ) ||
+				currentRoute.startsWith( '/checkout/jetpack' );
+			const noMasterbarForRoute = isJetpackLogin || currentRoute === '/me/account/closed';
+			const noMasterbarForSection = [ 'signup', 'jetpack-connect' ].includes( sectionName );
+			const masterbarIsHidden =
+				! masterbarIsVisible( state ) ||
+				noMasterbarForSection ||
+				noMasterbarForRoute ||
+				isWpMobileApp() ||
+				isWcMobileApp();
+			const isJetpackMobileFlow = 'jetpack-connect' === sectionName && !! retrieveMobileRedirect();
+			const isJetpackWooCommerceFlow =
+				[ 'jetpack-connect', 'login' ].includes( sectionName ) &&
+				'woocommerce-onboarding' === currentQuery?.from;
+			const isJetpackWooDnaFlow =
+				[ 'jetpack-connect', 'login' ].includes( sectionName ) &&
+				wooDnaConfig( currentQuery ).isWooDnaFlow();
+			const oauth2Client = getCurrentOAuth2Client( state );
+			const wccomFrom = currentQuery?.[ 'wccom-from' ];
+			const isEligibleForJITM = [
+				'home',
+				'stats',
+				'plans',
+				'themes',
+				'plugins',
+				'comments',
+			].includes( sectionName );
+			const sidebarIsHidden = ! secondary || isWcMobileApp();
+			const chatIsDocked = ! [ 'reader', 'theme' ].includes( sectionName ) && ! sidebarIsHidden;
+
+			const isEditor = getSectionName( state ) === 'gutenberg-editor';
+			const isCheckout = getSectionName( state ) === 'checkout';
+
+			const userAllowedToHelpCenter =
+				( ( isEditor && config.isEnabled( 'editor/help-center' ) ) ||
+					( isCheckout && config.isEnabled( 'checkout/help-center' ) ) ||
+					( ! isEditor && ! isCheckout && config.isEnabled( 'calypso/help-center' ) ) ) &&
+				shouldShowHelpCenterToUser( getCurrentUserId( state ) );
```

Comment:
I don't think we should hold checkout different anymore. 
```suggestion
			const userAllowedToHelpCenter =
				config.isEnabled( 'calypso/help-center' ) &&
				shouldShowHelpCenterToUser( getCurrentUserId( state ) );
```